src/Microsoft.VisualStudio.Threading/JoinableTaskFactory.cs

```diff
@@ -934,7 +934,7 @@ private void OnCompleted(Action continuation, bool flowExecutionContext)
                         // if the awaiter has been copied (since it's a struct), each copy of the awaiter
                         // points to the same registration. Without this we can have a memory leak.
                         var registration = this.cancellationToken.Register(
-                            flowExecutionContext ? SafeCancellationAction : UnsafeCancellationAction,
+                            (flowExecutionContext ? SafeCancellationAction : UnsafeCancellationAction)!,
```

Comment:
💭 I lean towards this, even though it's really the same:

```suggestion
                            flowExecutionContext ? SafeCancellationAction! : UnsafeCancellationAction!,
```

Reply:
It's not the same. Your change would reintroduce the warning. I believe it's another case of the `Action<object>` vs. `Action<object?>` rather than that individual values that might be null. I suspect the C# compiler doesn't apply the `!` such that it knows what I'm trying to accomplish except when applied as a whole.

Reply:
I wrapped this in the same helper method.